snap/hooks/install

```diff
@@ -2,32 +2,33 @@
 
 set -eux
 
-# Create the config file that is used by pbm
-PBM_CONFIG_FILE="${SNAP_COMMON}/pbm_config.yaml"
+# Create the necessary parent directories 
+mkdir -p "${SNAP_DATA}/etc/pbm/"
+mkdir -p "${SNAP_DATA}/etc/mongod/"
+mkdir -p "${SNAP_COMMON}/var/lib/mongod/db"
+mkdir -p "${SNAP_COMMON}/var/log/mongod/"
+
+# Create the config file that is used by pbm along with the directories
+PBM_CONFIG_FILE="${SNAP_DATA}/etc/pbm/pbm_config.yaml"
 touch "${PBM_CONFIG_FILE}"
 
-# the pbm-agent daemon is not run as root meaning it does not have access write to SNAP_DATA, so
-# we need to create a child dir and give it access.
-mkdir -p "${SNAP_DATA}/pbm-agent-dir"
-
 # Copy over the mongod.conf and create needed directories/files
-MONGO_CONFIG_FILE="${SNAP_COMMON}/mongod.conf"
+MONGO_CONFIG_FILE="${SNAP_DATA}/etc/mongod/mongod.conf"
 echo "configuration file does not exist."
 echo "copying default config to ${MONGO_CONFIG_FILE}"
-cp -r ${SNAP}/etc/mongod.conf ${SNAP_COMMON}
+cp -r ${SNAP}/etc/mongod.conf ${MONGO_CONFIG_FILE}
 
 # Creating various files/directories in snap folders
 # This is required as the snap_daemon user doesnt have permission
 # to write to those directories
-mkdir -p ${SNAP_COMMON}/db
-touch ${SNAP_DATA}/mongod.log
-touch ${SNAP_DATA}/mongod.pid
-
-
-sed -i "s/fork: true/fork: false/g" $SNAP_COMMON/mongod.conf
-sed -i "s:/var/log/mongodb/mongod.log:$SNAP_DATA/mongod.log:g" $SNAP_COMMON/mongod.conf
-sed -i "s:/var/lib/mongodb:$SNAP_COMMON/db:g" $SNAP_COMMON/mongod.conf
-sed -i "s:/var/run/mongod.pid:$SNAP_DATA/mongod.pid:g" $SNAP_COMMON/mongod.conf
+touch ${SNAP_COMMON}/var/log/mongod/mongod.log
+touch ${SNAP_COMMON}/var/lib/mongod/mongod.pid
```

Comment:
is there any reason why pid file should be preserved in common dir?

Reply:
I wasn't sure about this, but my reasoning was that according [to the spec document](https://docs.google.com/document/d/1MyT4Wt_-vgPwhwEjC-oeHRdLYrZWaQzFGTwclT7hD1U/edit?usp=sharing):
> data - the core workload data for the service running
For Kafka, this will be the topics, metadata, offsets etc
For ZooKeeper, this will be the state snapshots
The snap directory for this should $SNAP_COMMON/var/lib/COMPONENT/

I figured the `pid` was `metadata` that was relevant to the workload, and hence should go in `$SNAP_COMMON/var/lib/COMPONENT/`. But I wasn't entirely sure 😅  is there a more appropriate place?

Reply:
Ah I misunderstood the docs, it is better in conf files i.e. `$SNAP_DATA/etc/COMPONENT`- but lets figure out if we need this first